File: PI-Cloud-Backend/src/main/java/tn/esprit/peakwell/services/ReactionService.java
package tn.esprit.peakwell.services;

import tn.esprit.peakwell.dto.ReactionDTO;
import tn.esprit.peakwell.entities.Article;
import tn.esprit.peakwell.entities.Reaction;
import java.util.List;
import java.util.Map;

public interface ReactionService {
    void toggleReaction(Long articleId, String type, String userIdentifier);
    List<ReactionDTO> getReactionsByArticle(Long articleId);
    Map<String, Long> getReactionCounts(Long articleId);
}

File: PI-Cloud-Frontend/src/app/services/reaction.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Reaction, ReactionCount } from '../models/reaction.model';

@Injectable({
  providedIn: 'root'
})
export class ReactionService {
  private apiUrl = 'http://localhost:8090/peakwell/api/reactions';

  constructor(private http: HttpClient) { }

  toggleReaction(articleId: number, type: string, userIdentifier: string): Observable<void> {
    return this.http.post<void>(
      `${this.apiUrl}/article/${articleId}`,
      { type, userIdentifier }
    );
  }

  getReactionCounts(articleId: number): Observable<{ [key: string]: number }> {
    return this.http.get<{ [key: string]: number }>(
      `${this.apiUrl}/article/${articleId}/count`
    );
  }

  getReactions(articleId: number): Observable<Reaction[]> {
    return this.http.get<Reaction[]>(
      `${this.apiUrl}/article/${articleId}`
    );
  }
}
